src/Controller/Carrinho/CarrinhoApiController.php

```diff
@@ -21,27 +22,64 @@ public function carrinho(
         $idCliente,
         ClienteRepository $clienteRepository,
         CarrinhoRepository $carrinhoRepository,
+        ItemRepository $itemRepository,
         Security $security
     ): JsonResponse
     {
         $cliente = $clienteRepository->find($idCliente);
-        $carrinhoBusca = $carrinhoRepository->findOneBy(['cliente' => $cliente]);
+        $carrinhoBusca = $carrinhoRepository->buscar($cliente);
```

Comment:
podemos buscar por outras coisas
```suggestion
        $carrinhoBusca = $carrinhoRepository->buscarPorCliente($cliente);
```

Reply:
tem que olhar qual Carrinho está vindo, pode ser que tem mais Carrinhos para aquele Cliente e pode vir o errado, então seria legal olhar o status do Carrinho aqui também ao considerar a busca...